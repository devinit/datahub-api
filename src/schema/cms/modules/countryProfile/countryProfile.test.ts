import {getCountryProfilePageData} from '.';
import * as prettyFormat from 'pretty-format';

describe('country profile module tests', () => {
    // TOFIX: skipped coz it fails over network request issue
    it('should return page data for a country slug', async () => {
        const pageData = await getCountryProfilePageData('afghanistan');
        expect(pageData.length).toBeGreaterThan(2);
        expect(prettyFormat(pageData)).toMatchSnapshot();
    }, 100000);
});
