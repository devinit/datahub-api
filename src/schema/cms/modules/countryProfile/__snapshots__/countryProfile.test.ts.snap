// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`country profile module tests should return page data for a country slug 1`] = `
"Array [
  Object {
    \\"id\\": \\"introduction\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"Explore this in-depth profile of Afghanistan to find out overall levels of poverty, income distribution, division of wealth and more. Discover how national and sub-national revenue is generated\\",
  },
  Object {
    \\"id\\": \\"overview-ctry-poorest-people\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MANY OF THE POOREST PEOPLE GLOBALLY LIVE IN Afghanistan?\\",
  },
  Object {
    \\"id\\": \\"overview-ctry-resources\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT RESOURCES ARE AVAILABLE?\\",
  },
  Object {
    \\"id\\": \\"overview-govt-spentperperson\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH DOES THE GOVERNMENT SPEND PER PERSON?\\",
  },
  Object {
    \\"id\\": \\"poverty-reduction-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"IS POVERTY REDUCING OVER TIME?\\",
  },
  Object {
    \\"id\\": \\"poverty-depth-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW DEEP IS POVERTY?\\",
  },
  Object {
    \\"id\\": \\"income-distribution-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW IS INCOME DISTRIBUTED?\\",
  },
  Object {
    \\"id\\": \\"popn-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE POPULATION?\\",
  },
  Object {
    \\"id\\": \\"popn-distribution\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE URBAN VS RURAL SPLIT?\\",
  },
  Object {
    \\"id\\": \\"popn-distribution-age-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE AGE PROFILE?\\",
  },
  Object {
    \\"id\\": \\"totalrevenue-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHAT IS THE TOTAL REVENUE OF CTRY?\\",
  },
  Object {
    \\"id\\": \\"govt-revenue-external\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"HOW MUCH GOVERNMENT REVENUE COMES FROM EXTERNAL GRANTS (AID)?\\",
  },
  Object {
    \\"id\\": \\"ctry-expenditure\\",
    \\"meta\\": \\"Recurrent Expenditure\\",
    \\"narrative\\": \\"Capital Expenditure\\",
    \\"title\\": \\"HOW IS SPENDING ALLOCATED?\\",
  },
  Object {
    \\"id\\": \\"diff-providers-priorities\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"How effectively are donors targeting the countries with the highest numbers of people in poverty and the least domestic public resources to address it? This chart allows you to select a provider of aid to see how it allocates aid across countries\\",
    \\"title\\": \\"Different providers, different priorities\\",
  },
  Object {
    \\"id\\": \\"resources-able-to-meet-needs\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"The proportion of people living in extreme poverty worldwide has more than halved since 2000, yet progress is varied across countries. Although domestic public resources are an important resource to drive efforts to end poverty, they remain scarce in many countries where poverty is high\\",
    \\"title\\": \\"Are domestic public resources able to meet the needs of the poorest people?\\",
  },
  Object {
    \\"id\\": \\"Unbundling aid\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Explore and compare funding priorities for official development assistance\\",
    \\"title\\": \\"Unbundling aid\\",
  },
  Object {
    \\"id\\": \\"aid-allocated-ctry\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"AS A SHARE OF GNI, HOW MUCH AID IS ALLOCATED TO Country?\\",
  },
  Object {
    \\"id\\": \\"mixture-of-resources\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"\\",
    \\"title\\": \\"WHATâ€™S THE MIX OF RESOURCES?\\",
  },
  Object {
    \\"id\\": \\"Unbundling aid\\",
    \\"meta\\": \\"\\",
    \\"narrative\\": \\"Explore and compare funding priorities for official development assistance  Using this visualisation\\",
    \\"title\\": \\"Unbundling aid\\",
  },
]"
`;
